modules/services/searching/src/main/java/org/fao/geonet/searching/util/ElasticSearchProxy.java

```diff
@@ -369,33 +301,60 @@ private void handleRequest(HttpSession httpSession,
         }
 
         try {
-          searchResponseProcessor
-              .processResponse(httpSession, streamFromServer, streamToClient, userInfo,
-                  selectionBucket,
-                  addPermissions);
+          String acceptHeader = request.getHeader("Accept");
+          Class<? extends SearchResponseProcessor> responseProcessorClass =
+              RESPONSE_PROCESSOR.get(acceptHeader);
+          if (responseProcessorClass == null) {
+            throw new UnsupportedOperationException(String.format(
+                "No response processor configured for '%s'. Use one of %s.",
+                acceptHeader, RESPONSE_PROCESSOR.keySet().toArray()));
+          }
+
+          SearchResponseProcessor responseProcessor =
+              applicationContext.getBean(responseProcessorClass);
+          if (responseProcessor == null) {
+            throw new UnsupportedOperationException(String.format(
+                "No response processor bean found for '%s'.",
+                acceptHeader));
+          }
+
+          responseProcessor.processResponse(
+              httpSession,
+              streamFromServer, streamToClient,
+              userInfo, selectionBucket, addPermissions);
           streamToClient.flush();
         } finally {
           IOUtils.closeQuietly(streamFromServer);
         }
       } catch (Exception ex) {
         ex.printStackTrace();
+
+        throw new Exception(
+            String.format(
+                "Failed to connect to index at URL %s. %s",
+                esUrl, ex.getMessage()), ex);
       } finally {
         connectionWithFinalHost.disconnect();
       }
     } catch (IOException e) {
       // connection problem with the host
       e.printStackTrace();
 
-      throw new Exception(
-          String.format("Failed to request Es at URL %s. "
-                  + "Check Es configuration.", esUrl),
-          e);
+      throw new Exception(String.format(
+          "Failed to request index at URL %s. Check configuration.",
+          esUrl), e);
     }
   }
 
+  private boolean isSearch(HttpServletRequest request) {
```

Comment:
Can this method return `false`? In https://github.com/geonetwork/geonetwork-microservices/pull/19/files#diff-cfac1ab081c02d15be888f5d96ac3b90c62abf720af23167c64540c97c6e368cR304 if the `Accept` header is not related to a `RESPONSE_PROCESSOR` throws an exception.

I guess is more for future extensions?

Reply:
We used to have the possibility to call other ES API endpoints. But here we are probably limiting to search only so this is something we can drop.